library/src/com/google/maps/android/kml/KmlRenderer.java

```diff
@@ -890,11 +935,13 @@ public GroundOverlayImageDownload(String groundOverlayUrl) {
          */
         @Override
         protected Bitmap doInBackground(String... params) {
+            //Note: Doing this "URI uri = new URI(mGroundOverlayUrl);" doesn't work because it will always throw a syntax exception if there are spaces.
+            //If it's not a relative URL, then try to load from the world wide web.
             try {
                 return BitmapFactory
-                        .decodeStream((InputStream) new URL(mGroundOverlayUrl).getContent());
+                        .decodeStream((InputStream) new URL(mGroundOverlayUrl).getContent(), null, mBitmapOptions);
             } catch (MalformedURLException e) {
-                return BitmapFactory.decodeFile(mGroundOverlayUrl);
+                return BitmapFactory.decodeFile(mDirectoryName + "/" + mModifiedUrl, mBitmapOptions);
```

Comment:
`new File(mDirectoryName, mGroundOverlayUrl).getPath()` is much cleaner than manually concatenating the file path. Also avoids the need for `mModifiedUrl`.